julia

JuliaKovalchook opened the pull request.

JuliaKovalchook committed:
Update laboratory1.sql

JuliaKovalchook committed:
Update laboratory1.sql

Alexandr-gw commented on km62/Kovalchuk_Yuliia/laboratory1.sql:
```diff
@@ -1,2 +1,226 @@
+Людина купляє певну марку телефону.
+4 бали
+
+---------------------------------------------------------------------------*/
```
There should be tables - 'Human' and 'Phone' with many to many connection, because one person can own two different marks of phone and phone's which made by one company can be bought by many different people's. So table 'Human' and 'Phone' should have reference to each other's. And we also need the auxiliary table 'Orders' which will contain info about orders.
# Human
1) person_id - number(20) primary key not null,
2) first_name - varchar2(30) not null
Regexp - ^[A-Z][a-z]{1,29}$
3) second_name - varchar2(30) not null
Regexp - ^[A-Z][a-z]{1,29}$
4) country - varchar2(100) not null
Regexp - ^([A-Z][a-z]+)+$
# Phone
1) phone_imei -number(15) primary key not null,
2) owner_id - number(20) not null,
3) phone_model - varchar2(20) not null,
Regexp - ^[A-Z][a-z]{19}$ 
4) phone_mark - varchar2(15) not null,
Regexp - ^[A-Z][a-z]{14}$ 
# Orders
1) owner_id - number(20) primary key not null,
2) order_num -number(15) not null,
3) order_date - number(10) not null,
4) phone_imei -number(15) primary key not null,

Alexandr-gw commented on km62/Kovalchuk_Yuliia/laboratory1.sql:
```diff
@@ -1,2 +1,226 @@
+Яка ціна найдорожчого товару, що нікому не було продано?
+Виконати завдання в Алгебрі Кодда. 
+4 бали
+---------------------------------------------------------------------------*/
```
SQL request:

SELECT prod_price
FROM products, orderitems
MINUS
SELECT prod_price 
FROM products, orderitems
WHERE orderitems.prod_id = Products.prod_id ;

Alexandr-gw commented on km62/Kovalchuk_Yuliia/laboratory1.sql:
```diff
@@ -1,2 +1,226 @@
+Виконати завдання в SQL. 
+4 бали
+
+---------------------------------------------------------------------------*/
```
Request in algebra Kodda:

PROJECT Customers WHERE cust_id in(PROJECT orders WHERE order_num in (POJECT orderitems WHERE item_price in(PROJECT orderitems{max(item_price)}){order_num}){cust_id})
{Customers RENAME cust_name AS Customer_name}

Alexandr-gw commented on km62/Kovalchuk_Yuliia/laboratory1.sql:
```diff
@@ -1,2 +1,226 @@
+--Код відповідь:
+
+
+SELECT USERS_NAME,
```
Here should be name of customer where CUST.cust_id  same in PROD.cust_id and max(PROD.prod_price)

JuliaKovalchook committed:
Update laboratory1.sql

JuliaKovalchook committed:
Update laboratory1.sql

JuliaKovalchook committed:
Update laboratory1.sql

JuliaKovalchook committed:
Update laboratory1.sql

JuliaKovalchook commented:
rr

JuliaKovalchook committed:
csv_export.py

import csv

import cx_Oracle

connection = cx_Oracle.connect("system", "uuuliakov1607199por", "xe")
 
cursor_shop= connection.cursor()
 
cursor_shop.execute("""
SELECT
    TRIM(shop_id) as shop_id,
    TRIM(shop_name) as shop_name,
    TRIM(shopt_location) as shopt_location
FROM
    shop""")
 
 
 
for shop_id, shop_name, shopt_location in cursor_shop:
 
    with open("shop_"+shop_id+".csv", "w", newline="") as file:
        writer = csv.writer(file)
 
        writer.writerow(["ID", shop_id])
        writer.writerow(["Name", shop_name])
        writer.writerow(["Location", shop_locatione])
 
        cursor_phone = connection.cursor()
 
        query = """
                    SELECT
                        *
                    FROM
                         phone
                    WHERE TRIM(shop_id) = :id"""
 
        cursor_phone.execute(query, id = shop_id)
        writer.writerow([])
        writer.writerow(["phone_id", "phone number", "phone price", "phone_marka"])
        for phone_row in cursor_phone:
            writer.writerow(phone_row)
 
cursor_shop.close()

JuliaKovalchook committed:
csv_import.py

JuliaKovalchook committed:
Update and rename km62/Kovalchuk_Yuliia/laboratory1.sql to  км62/ Ковальчук_Юлія/laboratory1.sql

JuliaKovalchook committed:
csv_export.py

JuliaKovalchook committed:
Rename laboratory4.sql to laboratory4_hw.sql

JuliaKovalchook committed:
Update laboratory4.sql

JuliaKovalchook committed:
Delete laboratory4_hw.sql

JuliaKovalchook commented:
https://plot.ly/dashboard/JuliaKovalchuk:6/view#/

JuliaKovalchook commented:
https://plot.ly/dashboard/JuliaKovalchuk:6/view#/

JuliaKovalchook commented:
https://plot.ly/dashboard/JuliaKovalchuk:6/view#/

JuliaKovalchook commented:
![erd](https://user-images.githubusercontent.com/22946872/41127375-64a7fdc0-6ab3-11e8-95ad-b90ccc035df6.png)
